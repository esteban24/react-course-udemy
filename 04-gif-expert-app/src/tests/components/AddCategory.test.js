import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"

describe('AddCategory.js test suite', () => {
    const wrapper = shallow(<AddCategory setCategories={ () => console.log('set categories') }/>);
    
    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change input form field', () => {
        const input = wrapper.find('input');

        input.simulate('change', { target: { value: 'Salu2' } });

        const actualValue = wrapper.props('inputValue').children[0].props.value;

        expect(actualValue).toBe('Salu2');
    })    
})
