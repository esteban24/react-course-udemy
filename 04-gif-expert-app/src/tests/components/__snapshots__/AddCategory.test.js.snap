// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory.js test suite should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    AÃ±adir
  </button>
</form>
`;
